File: frontend/src/lib/api/vision.ts
import apiClient from './client';

export interface ImageAnalysisResponse {
  analysis_id: number;
  image_type: string;
  findings: string;
  full_analysis: string;
  severity: string;
  confidence: number;
  recommendations: string;
  metadata: Record<string, any>;
  created_at: string;
}

export interface XRayAnalysisResponse {
  analysis_id: number;
  analysis: string;
  findings: string[];
  severity: string;
  confidence: number;
  differential_diagnosis: string[];
  recommendations: string;
  urgent_flags: string[];
}

export interface XRayAnalysisParams {
  file: File;
  symptoms?: string;
  age?: number;
  gender?: string;
  medicalHistory?: string;
}

export const visionAPI = {
  analyzeImage: async (
    file: File,
    imageType: string,
    patientContext?: string,
    enhanceImage: boolean = true,
    patientId?: number
  ): Promise<ImageAnalysisResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('image_type', imageType);
    if (patientContext) formData.append('patient_context', patientContext);
    formData.append('enhance_image', String(enhanceImage));
    if (patientId) formData.append('patient_id', String(patientId));

    const response = await apiClient.post('/vision/analyze', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },


  analyzeXRay: async (params: XRayAnalysisParams): Promise<XRayAnalysisResponse> => {
    const formData = new FormData();
    formData.append('file', params.file);
    if (params.symptoms) formData.append('symptoms', params.symptoms);
    if (params.age) formData.append('age', String(params.age));
    if (params.gender) formData.append('gender', params.gender);
    if (params.medicalHistory) formData.append('medical_history', params.medicalHistory);

    const response = await apiClient.post('/vision/xray/analyze', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
      timeout:90000
    });
    return response.data;
  },

  getAnalysis: async (analysisId: number): Promise<ImageAnalysisResponse> => {
    const response = await apiClient.get(`/vision/analysis/${analysisId}`);
    return response.data;
  },

  getHistory: async (limit: number = 10, imageType?: string) => {
    const params: any = { limit };
    if (imageType) params.image_type = imageType;

    const response = await apiClient.get('/vision/history', { params });
    return response.data;
  },
};

File: backend/app/schemas/vision.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, Dict, List, Any
from datetime import datetime


class ImageUploadResponse(BaseModel):
    message: str
    image_id: int
    analysis_id: Optional[int] = None
    
    model_config = ConfigDict(from_attributes=True)


class ImageAnalysisRequest(BaseModel):
    image_type: str = Field(
        ..., 
        pattern="^(chest_xray|skin_lesion|ct_scan|mri|general)$",
        description="Type: chest_xray, skin_lesion, ct_scan, mri"
    )
    patient_context: Optional[str] = Field(
        None, 
        max_length=2000,
        description="Patient symptoms or relevant medical history"
    )
    enhance_image: bool = Field(True, description="Apply CLAHE enhancement before analysis")


class ImageAnalysisResponse(BaseModel):
    analysis_id: int
    image_type: str
    findings: str
    full_analysis: str
    severity: str = Field(..., pattern="^(normal|mild|moderate|severe|critical)$")
    confidence: float = Field(..., ge=0, le=1)
    recommendations: str
    metadata: Dict[str, Any]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class XRayAnalysisRequest(BaseModel):
    symptoms: Optional[str] = None
    age: Optional[int] = Field(None, ge=0, le=120)
    gender: Optional[str] = None
    medical_history: Optional[str] = None



class XRayAnalysisResponse(BaseModel):
    analysis_id: int
    analysis: str
    findings: List[str]
    severity: str
    confidence: float
    differential_diagnosis: List[str]
    recommendations: str
    urgent_flags: List[str]

    model_config = ConfigDict(from_attributes=True)